android/build.gradle

```diff
@@ -1,3 +1,31 @@
+import groovy.json.JsonSlurper
+
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
+def warnNoAutoInit() {
+    logger.warn("Couldn't auto-initialize the Heap library; manually initialize it by following the instructions at https://docs.heap.io/docs/react-native.")
+}
+
+def warnDisabled() {
+    logger.warn("Not auto-initializing the Heap library because it was manually disabled; set `enabled` to `true` in `heap.json` to enable auto-initialization.")
+}
+
+def heapAppId() {
+    def heapConfigFilename = 'heap.json'
+    def file = rootProject.file("../${heapConfigFilename}")
+    if (!file.exists()) { return warnNoAutoInit() }
+
+    def jsonSlurper = new JsonSlurper()
+    def data = jsonSlurper.parseText(file.text.trim())
+
+    if (!data instanceof Map || !data.appId instanceof String) { return warnNoAutoInit() }
```

Comment:
I think this gets a bit funny to read... might be nice to phrase this in the positive rather than the negative (especially considering there will just be a single `return` inside the conditional block.

Which, BTW, I've always followed style guidelines to break those blocks into three lines (i.e. "{\nFN_BODY\n}") for readability, except in a few rare circumstances.  It seems weightier on the page, and draws my eye into it.

That being said, I'm a big fun of the `guard let ...` construct common in Swift, where an early abort is just done on a single line... so you commonly see things like:
```
guard let neverNullVal = possiblyNullVal else { return }
```
... which is also nice because you've got a variable forever after that line that's guaranteed to be good and you don't have to worry about it.  And in that case, I think the braces on the same line as the conditional makes things neat and tidy, so go figure.

What do you think?